python/servo/command_base.py

```diff
@@ -852,6 +852,11 @@ def run_cargo_build_like_command(
             if self.config["build"]["debug-mozjs"] or debug_mozjs:
                 features.append("debugmozjs")
 
+            if path.join(self.context.topdir, "ports", "winit", "Cargo.toml") not in cargo_args:
+                cargo_args += ["--no-default-features"]
+                features.append("webdriver")
+                features.append("max_log_level")
+
```

Comment:
This is the most dirty part, that also probably causes collision between `mach build -d` and `cargo build` clash so it always rebuild if run combined, probably due to `--no-default-feature` even though at the end all same features are enabled.

Reply:
I think this should be fine since mixing commands should be rare.

Reply:
Hrm. A great goal here is that the defaults for rust-analyzer don't cause massive rebuilds, so mixing commands seems within the realm.